docs/components/discovery.md

```diff
@@ -33,7 +38,7 @@ class MyClass
     protected $httpClient;
 
     /**
-     * @param HttpClient|null $httpClient Client to do HTTP requests, if not set, autodiscovery will be used to find a HTTP client.
+     * @param HttpClient|null $httpClient
```

Comment:
i would want my code to have this comment. IDEs don't show you the
constructor code, but they show the phpdoc of the constructor, so this
would help understand what the semantics of the constructor are.